components/lab-job-scheduler/backend/src/lab_job_scheduler/app.py

```diff
@@ -0,0 +1,232 @@
+
+    return cached_scheduled_jobs[project_id][job_id]
+
+
+@app.delete(
+    "/project/{project_id}/schedules",
+    summary="Deletes all job schedules",
+    status_code=status.HTTP_200_OK,
+    responses={**CREATE_RESOURCE_RESPONSES},
+)
+def delete_schedules(
+    project_id: str,
+    component_manager: ComponentOperations = Depends(get_component_manager),
+) -> Any:
+    db = component_manager.get_json_db_manager()
+    db.delete_json_collection(project_id=project_id, collection_id="schedules")
+    cached_scheduled_jobs[project_id] = {}
+
+
+@app.delete(
+    "/project/{project_id}/schedule/{job_id}",
+    summary="Deletes a job schedule",
+    status_code=status.HTTP_200_OK,
+    responses={**CREATE_RESOURCE_RESPONSES},
+)
+def delete_schedule(
+    project_id: str,
+    job_id: str,
+    component_manager: ComponentOperations = Depends(get_component_manager),
+) -> Any:
+    db = component_manager.get_json_db_manager()
+    db.delete_json_document(
+        project_id=project_id, collection_id="schedules", key=job_id)
+
+    if project_id in cached_scheduled_jobs and job_id in cached_scheduled_jobs[project_id]:
+        del cached_scheduled_jobs[project_id][job_id]
+
+
+@app.post(
+    "/project/{project_id}/schedule/{job_id}",
+    summary="Updates a job schedule",
+    status_code=status.HTTP_200_OK,
+    responses={**CREATE_RESOURCE_RESPONSES},
+)
+def update_schedule(
+    project_id: str,
+    job_input: ScheduledJobInput,
+    job_id: str,
+    component_manager: ComponentOperations = Depends(get_component_manager),
+) -> Any:
+    job = get_job_from_job_input(job_input)
+    db = component_manager.get_json_db_manager()
+    resp = db.update_json_document(project_id=project_id,
+                                   collection_id="schedules", key=job_id, json_document=json.dumps((job.dict())))
+
+    if project_id not in cached_scheduled_jobs:
+        cached_scheduled_jobs[project_id] = {}
+
+    cached_scheduled_jobs[project_id][job_id] = job
+    return resp
+
+
+@app.get(
+    "/executor/info",
+    summary="Display information about the executor",
+    status_code=status.HTTP_200_OK,
+    responses={**CREATE_RESOURCE_RESPONSES},
+)
+def executor_info() -> Any:
+    return {
+        "execution_frequency": JOB_INTERVAL,
+    }
+
+
+def get_all_scheduled_jobs_from_db(component_manager: ComponentOperations, project_id: str) -> List[ScheduledJob]:
+    """Returns all jobs from the database."""
+    db = component_manager.get_json_db_manager()
+    documents = db.list_json_documents(
+        project_id=project_id, collection_id="schedules"
+    )
+    return [ScheduledJob(**json.loads(document.json_value)) for document in documents]
+
+
+def get_job_from_job_input(job_schedule: ScheduledJobInput) -> ScheduledJob:
+    return ScheduledJob(
+        cron_string=job_schedule.cron_string,
+        job_input=job_schedule.job_input,
+        created=datetime.datetime.now().isoformat(),
+        job_id=str(uuid.uuid4()),
```

Comment:
Use the same function to generate the id that is also used in contaxy:
```
from contaxy.utils import id_utils
id_utils.generate_short_uuid()
```

Reply:
Resolved in 839b1c30aa0fb87be8805706d597dc5aa4f9c286